man/sd_bus_service_reconnect.c

```diff
@@ -92,6 +92,24 @@ static int on_disconnect(sd_bus_message *message, void *userdata, sd_bus_error *
   return 0;
 }
 
+/* Ensure the event loop exits with a clear error if acquiring the well-known service name fails */
+static int request_name_callback(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
+  if (sd_bus_message_is_method_error(m, NULL)) {
+    int r = sd_bus_message_get_errno(m);
+
+    if (r == ETIMEDOUT)
+      return 1; /* The bus is not available, try again later */
+
+    printf("Failed to request name: %s\n", strerror(r));
+    object *o = userdata;
+    check(sd_event_exit(*o->event, -r));
+
+    return -r;
```

Comment:
might as well return 1 here too. no reason to fail the callback if you asked the event loop to exit anyway